src/libs/actions/Task.js

```diff
@@ -0,0 +1,219 @@
+import Onyx from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as API from '../API';
+import * as ReportUtils from '../ReportUtils';
+import * as Report from './Report';
+import Navigation from '../Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+
+/**
+ * Clears out the task info from the store
+ */
+function clearOutTaskInfo() {
+    Onyx.set(ONYXKEYS.TASK, null);
+}
+
+/**
+ * Assign a task to a user
+ * Function title is createTask for consistency with the rest of the actions
+ * and also because we can create a task without assigning it to anyone
+ * @param {String} currentUserEmail
+ * @param {String} parentReportID
+ * @param {String} title
+ * @param {String} description
+ * @param {String} assignee
+ *
+ */
+
+function createTaskAndNavigate(currentUserEmail, parentReportID, title, description, assignee = '') {
+    // Create the task report
+    const optimisticTaskReport = ReportUtils.buildOptimisticTaskReport(currentUserEmail, assignee, parentReportID, title, description);
+
+    // Grab the assigneeChatReportID if there is an assignee and if it's not the same as the parentReportID
+    // then we create an optimistic add comment report action on the assignee's chat to notify them of the task
+    const assigneeChatReportID = lodashGet(ReportUtils.getChatByParticipants([assignee]), 'reportID');
+    let optimisticAssigneeAddComment;
+    if (assigneeChatReportID && assigneeChatReportID !== parentReportID) {
+        optimisticAssigneeAddComment = ReportUtils.buildOptimisticAddCommentReportAction(
+            `${currentUserEmail} has[created a task for you](tbd/r/${optimisticTaskReport.reportID}): ${title}`,
+        );
+        optimisticAssigneeAddComment.reportAction.message[0].taskReportID = optimisticTaskReport.reportID;
+    }
+
+    // Create the CreatedReportAction on the task
+    const optimisticTaskCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(optimisticTaskReport.reportID);
```

Comment:
Coming from [No activity yet appears in task item of search RHN.](https://github.com/Expensify/App/issues/26679#top)

We should have updated `getLastMessageTextForReport()` to show the task details in LHN description.